homeassistant/components/modbus/binary_sensor.py

```diff
@@ -59,7 +59,7 @@ class ModbusBinarySensor(BasePlatform, RestoreEntity, BinarySensorEntity):
     def __init__(self, hub: ModbusHub, entry: dict[str, Any], slave_count: int) -> None:
         """Initialize the Modbus binary sensor."""
         self._count = slave_count + 1
-        self._coordinator: DataUpdateCoordinator[Any] | None = None
+        self._coordinator: DataUpdateCoordinator[list | None] | None = None
```

Comment:
Below is was specified as `list[str] | None` any reason the list isn't more specific in this case?

Reply:
It's probably that, however the code does dynamic assignments which I didn't fully check. So it could also be some other type as `str`. To be safe, I just used the generic `list` here.

**Edit**: Looking a bit further it seems, my instinct was correct. If I had to guess, I would probably say it should be `list[int] | None`.
https://github.com/home-assistant/core/blob/72671b93d29f5bbe10e18cffb3c89050a587ee40/homeassistant/components/modbus/binary_sensor.py#L148-L150

It would be nice to use the full type, but it doesn't really matter much here tbh.
Maybe @janiversen knows more?

Reply:
list[int] | None is the correct type to use